### PR TITLE
Integrate Twilio and send an SMS confirmation message when application is signed

### PR DESCRIPTION
**Pull request by mpidcock**

### CODE REVIEW

File: src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java [hunk 1]
```diff
@@ -22,12 +22,31 @@ public ConfirmationMessage(Submission submission, Transmission transmission) {
...
   public Email renderEmail() {
-    var applicationNumber = SubmissionUtilities.getFormattedConfirmationNumber(transmission.getConfirmationNumber());
-    String subject = "Test confirmation email";
+    var confirmationNumber = SubmissionUtilities.getFormattedConfirmationNumber(transmission.getConfirmationNumber());
+    var applicantFullName = SubmissionUtilities.applicantFullName(submission);
+    String subject = "Application Submitted for P-EBT 4.0";
+    String body = """
+      <html>
+        <body>
+          <p>Dear %s</p>
```
Comment (alaroia-cfa) [inline src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java:31]: It's not super duper important, but is this %s HTML-injectable? It's probably smart to escape this and/or use a Thymeleaf template so we get auto-escaping.

File: src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java [hunk 2]
```diff
@@ -22,12 +22,31 @@ public ConfirmationMessage(Submission submission, Transmission transmission) {
...
 
-    String emailBody = "<html><body><p>Your confirmation number is: <br>%s</p></body></html>".formatted(
-      applicationNumber
-    );
-    return new Email(subject, emailBody);
+  @Override
+  public Sms renderSms() {
+    var confirmationNumber = SubmissionUtilities.getFormattedConfirmationNumber(transmission.getConfirmationNumber());
+    String body = """
+      Thank you for submitting the application for P-EBT benefits for the 2022-2023 school year. You will hear back within 2-4 weeks. Your application number is %s.\s
```
Comment (alaroia-cfa) [inline src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java:47]: Nice, TIL \s https://docs.oracle.com/en/java/javase/17/text-blocks/index.html

File: src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java (hunk 1 above)
Comment (alaroia-cfa) [reply src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java:31]: It's usually best to do "Dear %s," i.e. have a comma in the "Dear..." line. Not a blocker.

**Review by alaroia-cfa — APPROVED** [review]
For the log levels, are we running production in log level info? If so, I think INFO is a fine level for these log messages.

This code looks simple and nice. It would be good if we could cover this with a test - maybe the journey test can mock Twilio/Mailgun and prove a message actually gets sent. Could be something to discuss, but not a blocker.

One strong suggestion for HTML escaping (can be done in a follow-up).

Looks good! Approving as-is because the escaping request isn't security-critical.

File: src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java (hunk 1 above)
Comment (mpidcock) [reply src/main/java/org/homeschoolpebt/app/submission/messages/ConfirmationMessage.java:31]: Not injectable as far as I can tell. `formatted` builds the string right there to pass to the mailgun client.

I agree with the comma, but I went with what the copy said. I felt it was too minor to call out

**Pull request merged**